pymc3/distributions/mixture.py

```diff
@@ -0,0 +1,169 @@
+            return tt.stack([comp_dist.mean for comp_dist in self.comp_dists],
+                            axis=1)
+
+    def _comp_modes(self):
+        try:
+            return self.comp_dists.mode
+        except AttributeError:
+            return tt.stack([comp_dist.mode for comp_dist in self.comp_dists],
+                            axis=1)
+
+    def _comp_samples(self, point=None, size=None, repeat=None):
+        try:
+            samples = self.comp_dists.random(point=point, size=size, repeat=repeat)
+        except AttributeError:
+            samples = np.column_stack([comp_dist.random(point=point, size=size, repeat=repeat)
+                                       for comp_dist in self.comp_dists])
+
+        return np.squeeze(samples)
+        
+    def logp(self, value):
+        w = self.w
+        
+        return bound(logsumexp(tt.log(w) + self._comp_logp(value), axis=-1).sum(),
+                     w >= 0, w <= 1, tt.allclose(w.sum(axis=-1), 1))
+
+    def random(self, point=None, size=None, repeat=None):
+        def random_choice(*args, **kwargs):
+            w = kwargs.pop('w')
+            w /= w.sum(axis=-1, keepdims=True)
+            k = w.shape[-1]
+
+            if w.ndim > 1:
+                return np.row_stack([np.random.choice(k, p=w_) for w_ in w])
+            else:
+                return np.random.choice(k, p=w, *args, **kwargs)
+
+        w = draw_values([self.w], point=point)
+
+        w_samples = generate_samples(random_choice,
+                                     w=w,
+                                     broadcast_shape=w.shape[:-1] or (1,),
+                                     dist_shape=self.shape,
+                                     size=size).squeeze()
+        comp_samples = self._comp_samples(point=point, size=size, repeat=repeat)
+
+        if comp_samples.ndim > 1:
+            return np.squeeze(comp_samples[np.arange(w_samples.size), w_samples])
+        else:
+            return np.squeeze(comp_samples[w_samples])
+
+    
+class NormalMixture(Mixture):
+    R"""
+    Normal mixture log-likelihood
+
+    .. math:: f(x \mid w, \mu, \sigma^2) = \sum_{i = 1}^n w_i N(x \mid \mu_i, \sigma^2_i
+
+    ========  =======================================
+    Support   :math:`x \in \mathbb{R}`
+    Mean      :math:`\sum_{i = 1}^n w_i \mu_i`
+    Variance  :math:`\sum_{i = 1}^n w_i^2 \sigma^2_i`
+    ========  =======================================
+
+    Parameters
+    w : array of floats
+        w >= 0 and w <= 1
+        the mixutre weights
+    mu : array of floats
+        the component means
+    sd : array of floats
+        the component standard deviations
+    tau : array of floats
+        the component precisions
+    """
+    def __init__(self, w, mu, *args, **kwargs):
+        _, sd = get_tau_sd(tau=kwargs.pop('tau', None),
+                           sd=kwargs.pop('sd', None))
+        
+        super(NormalMixture, self).__init__(w, Normal.dist(mu, sd=sd),
```

Comment:
The simplicity to create a Normal Mixture here is really validating, nicely done.